Automattic/pocket-casts-ios
# Referrals: Switch to 2 months and make display of offer duration more flexible

| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This PR updates the referrals code to support a more flexible way to define offer duration.
At the moment it changes the code to an hard coded 2 month duration but this will be resolved using IAP info in the future.

| 1 | 2 | 3 |
| - | - | - |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-09-12 at 14 26 27](https://github.com/user-attachments/assets/99e48120-2c84-4f0f-b8d6-21ebd53172bb) | ![Simulator Screenshot - iPhone 15 Pro - 2024-09-12 at 14 19 06](https://github.com/user-attachments/assets/a7805bea-aa55-4e75-87a6-d86b2cc6a680) | ![Simulator Screenshot - iPhone 15 Pro - 2024-09-12 at 14 18 57](https://github.com/user-attachments/assets/02321ce9-9d6a-47d1-a4cf-7459dae6f572) |

## To test

1. Start the app from scratch
2. Ensure that you have the referrals FF enabled
3. Switch to the Profile view
4. Ensure that you have a Plus subscription (You can do it in settings -> developer)
5. Tap on the gift icon on the top
6. See if the guest pass duration is show correctly. ( It should be 2 months)
7. Tap on Close
8. Check that banner info for claiming a guest pass shows correctly
9. Tap on the banner for claiming guest pass
10. Check that the guest pass offer duration (2 months) and price after offer (39.99) is show correctly 


## Checklist
